### ISSUE TITLE
Temple progress not saved with autosave

### ISSUE THREAD

**Issue by TSEG6** [Mr-Wiseguy/Zelda64Recomp#59]
**Describe the bug**

**To Reproduce**
1. Go to snowhead temple
2. Do some of the temple
3. Wait for an auto save
4. Reload game
5. See issue

**Expected behavior**
After loading the autosave the dungeon progress should be saved.

**Screenshots**
Nothing would really work aside from being able to collect more than 15 stray fairies.

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Version: 1.0.0
 - CPU: Intel Core i7 12700KF
 - GPU: Geforce RTX 3070 ti

**Additional context**
I wanted to start the game back up after going to bed and the dungeon progress was reset but not my items or fairies.
This was also right after getting the fire arrows and exiting the room.

**Comment by zan7737** [comment]
I had the same issue today in Woodfall Temple.
After loading from an auto save I still had all the items I collected from the dungeon, however all the stray fairies and all previously opened chests I collected were back, despite already being counted as collected in the menu. The swamp room had blue water again, but Tatl still commented on how it smells like swamp. Collecting the same stray fairies again makes the counter go over the intended amount. 
I am using a Steam Deck.
I closed the game because I had to do something as well.
